src/core/server/saved_objects/service/lib/repository.ts

```diff
@@ -1896,6 +1908,48 @@ export class SavedObjectsRepository {
     return body;
   }
 
+  /**
+   * Returns a generator to help page through large sets of saved objects.
+   *
+   * The generator wraps calls to `SavedObjects.find` and iterates over
+   * multiple pages of results using `_pit` and `search_after`. This will
+   * open a new Point In Time (PIT), and continue paging until a set of
+   * results is received that's smaller than the designated `perPage`.
+   *
+   * Once you have retrieved all of the results you need, it is recommended
+   * to call `close()` to clean up the PIT and prevent Elasticsearch from
+   * consuming resources unnecessarily. This will automatically be done for
+   * you if you reach the last page of results.
```

Comment:
NIT: should we also document that `close()` will be called if the underlying call to `find` fails for any reason, and that it's not necessary to `try {} catch (e) {close}`?